packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -1400,82 +1404,83 @@ async function handler(
-            where: {
-              id: bookingSeat.id,
-            },
-            data: {
-              bookingId: newTimeSlotBooking.id,
-            },
-          }),
-        ]);
-      }
+        // Need to change the new seat reference and attendee record to remove it from the old booking and add it to the new booking
+        // https://stackoverflow.com/questions/4980963/database-insert-new-rows-or-update-existing-ones
+        if (seatAttendee?.id && bookingSeat?.id) {
+          await Promise.all([
+            await prisma.attendee.update({
+              where: {
+                id: seatAttendee.id,
+              },
+              data: {
+                bookingId: newTimeSlotBooking.id,
+              },
+            }),
+            await prisma.bookingSeat.update({
+              where: {
+                id: bookingSeat.id,
+              },
+              data: {
+                bookingId: newTimeSlotBooking.id,
+              },
+            }),
+          ]);
+        }
 
-      const copyEvent = cloneDeep(evt);
+        const copyEvent = cloneDeep(evt);
 
-      const updateManager = await eventManager.reschedule(copyEvent, rescheduleUid, newTimeSlotBooking.id);
+        const updateManager = await eventManager.reschedule(copyEvent, rescheduleUid, newTimeSlotBooking.id);
 
-      const results = updateManager.results;
+        const results = updateManager.results;
 
-      const calendarResult = results.find((result) => result.type.includes("_calendar"));
+        const calendarResult = results.find((result) => result.type.includes("_calendar"));
 
-      evt.iCalUID = Array.isArray(calendarResult?.updatedEvent)
-        ? calendarResult?.updatedEvent[0]?.iCalUID
-        : calendarResult?.updatedEvent?.iCalUID || undefined;
+        evt.iCalUID = Array.isArray(calendarResult?.updatedEvent)
+          ? calendarResult?.updatedEvent[0]?.iCalUID
+          : calendarResult?.updatedEvent?.iCalUID || undefined;
 
-      await sendRescheduledSeatEmail(copyEvent, seatAttendee as Person);
-      const filteredAttendees = originalRescheduledBooking?.attendees.filter((attendee) => {
-        return attendee.email !== bookerEmail;
-      });
-      await lastAttendeeDeleteBooking(originalRescheduledBooking, filteredAttendees, originalBookingEvt);
+        await sendRescheduledSeatEmail(copyEvent, seatAttendee as Person);
+        const filteredAttendees = originalRescheduledBooking?.attendees.filter((attendee) => {
+          return attendee.email !== bookerEmail;
+        });
+        await lastAttendeeDeleteBooking(originalRescheduledBooking, filteredAttendees, originalBookingEvt);
 
-      const foundBooking = await findBookingQuery(newTimeSlotBooking.id);
+        const foundBooking = await findBookingQuery(newTimeSlotBooking.id);
 
-      resultBooking = { ...foundBooking, seatReferenceUid: bookingSeat?.referenceUid };
+        resultBooking = { ...foundBooking, seatReferenceUid: bookingSeat?.referenceUid };
+      }
```

Comment:
Is this change intended?

Reply:
@joeauyeung

Reply:
I changed some of the logic flow. This changed how these lines were nested so the spaces were added by the linter.

Reply:
Are you sure @joeauyeung? Seems like that bracket moved way down. I'm just trying to grasp if this is intentional.

<img width="761" alt="image" src="https://github.com/calcom/cal.com/assets/3504472/f3148b49-8e19-478b-a45e-d15c218a1025">

Reply:
This change is intentional.